AstarVienna/ScopeSim
# Cleanup of Tests

## Changes made

- Add file `conftest.py`, which is recognized by pytest and used as a "global" location for any tests in ScopeSim.
- Add fixtures in that file to define mock locations and use those wherever we need mock files.
- Make names for mock files as explicit (absolute) as possible, instead of relying on `find_file` too much.
- Replace custom setup and teardown with fixtures.
- Proper patching of globals like `rc.__currsys__`.
- Try not to have tests influencing each other via globals.
- Cleanup of imports, remove unused stuff.
- Changes to `SystemDict`, add `UniqueList`, both will be moved to `astar_utils`
- Many small changes, formatting.

## Test guide

To have a single location where all the things applied here are written down, I added the file `howtotest.md` in the tests folder. This is mostly meant for all of us to have a place to look things up, but also intended to be part of a "contribution" guide for new contributors. The goal is to have unified standards in the test suite, and this file should be the place where those are defined. Of course, that file is not a holy scripture, and is meant to be expanded/updated when it makes sense.
